README.md
```r
library(nzcrash)
library(dplyr)
library(tidyr)
library(magrittr)
library(stringr)
library(ggplot2)
library(scales)
library(lubridate)
```

# nzcrash

This package redistributes [crash
statistics](http://www.nzta.govt.nz/resources/crash-analysis-system-data/)
already available from the New Zealand Transport Agency, but in a more
convenient form.

It's a large package (over 20 megabytes, compressed).

## Datasets

The `crashes` dataset describes most facts about a crash.  The datasets `causes`,
`vehicles`, and `objects_struck` describe facts that are in a many-to-one
relationship with crashes.  They can be joined to the `crashes` dataset by the
common `id` column.  The `causes` dataset can additionally be joined to the
`vehicles` dataset by the combination of the `id` and `vehicle_id` columns.
This is most useful when the resulting table is also joined to the `crashes`
dataset.

## Up-to-date-ness

The data was last scraped from the NZTA website on 2015-09-30.  At
that time, the NZTA had published data up to 2015-03-10.


```r
dim(crashes)
```

```
## [1] 540888     32
```

```r
dim(causes)
```

```
## [1] 888072      7
```

```r
dim(vehicles)
```

```
## [1] 979930      3
```

```r
dim(objects_struck)
```

```
## [1] 261276      3
```

## Accuracy

The [NZTA](http://www.transport.govt.nz/research/roadtoll/#5), doesn't agree with [itself](http://www.transport.govt.nz/research/roadtoll/annualroadtollhistoricalinformation/) about recent annual road tolls, and this dataset gives a third opinion.


```r
crashes %>% 
  filter(severity == "fatal") %>%
  group_by(year = year(date)) %>%
  summarize(fatalities = sum(fatalities))
```

```
## Source: local data frame [16 x 2]
## 
##     year fatalities
##    (dbl)      (int)
## 1   2000        462
## 2   2001        455
## 3   2002        405
## 4   2003        461
## 5   2004        435
## 6   2005        405
## 7   2006        393
## 8   2007        421
## 9   2008        366
## 10  2009        384
## 11  2010        375
## 12  2011        284
## 13  2012        308
## 14  2013        256
## 15  2014        279
## 16  2015         34
```

## Severity

Crashes categorised as "fatal", "serious", "minor" or "non-injury", based on the
casualties.  If there are any fatalities, then the crash is a "fatal" crash,
otherwise if there are any 'severe' injuries, the crash is a "serious" crash.

The definition of a 'severe' injury is not clear.

Minor and non-injury crashes are likely to be under-recorded since they often do
not involve the police, who write most of the crash reports upon which these
datasets are based.

A common mistake is to confuse the number of fatal crashes with the number of
fatalities.


```r
crashes %>% filter(severity == "fatal") %>% nrow
```

```
## [1] 5042
```

```r
sum(crashes$fatalities)
```

```
## [1] 5723
```

## Dates and times

Three columns of the `crashes` dataset describe the date and time of the crash
in the NZST time zone (Pacific/Auckland).

* `date` gives the date without the time
* `time` gives the time where this is available, and NA otherwise.  Times are
  stored as date-times on the first of January, 1970.
* `datetime` gives the date and time in one value when both are available, and
  NA otherwise.  `date` is always available, however `time` is not.

When aggregating by some function of the date, e.g. by year, then always start
from the `date` column unless you also need the time.  This ensures against
accidentally discounting crashes where a time is not recorded.


```r
crashes %>%
  filter(is.na(time)) %>%
  count(year = year(date)) %>%
  ggplot(aes(year, n)) +
  geom_line() +
  ggtitle("Crashes missing\ntime-of-day information")

crashes %>%
  filter(is.na(time)) %>%
  count(year = year(date)) %>%
  mutate(percent = n/sum(n)) %>%
  ggplot(aes(year, percent)) +
  geom_line() +
  scale_y_continuous(labels = percent) +
  ggtitle("Percent of crashes missing\ntime-of-day information")
```

![plot of chunk unnamed-chunk-5](figure/unnamed-chunk-5-1.png) ![plot of chunk unnamed-chunk-5](figure/unnamed-chunk-5-2.png) 

## Location coordinates

99.9% of
crashes have coordinates.  These have been converted from the NZTM projection to
the WGS84 projection for convenience with packages like `ggmap`.

Because New Zealand is tall and skinny, you can easily spot the main population
centres with a simple histogram.


```r
crashes %>%
  ggplot(aes(northing)) +
  geom_histogram(binwidth = .1)
```

![plot of chunk unnamed-chunk-6](figure/unnamed-chunk-6-1.png) 

## Vehicles

There can be many vehicles in one crash, so vehicles are recorded in a separate
`vehicles` dataset that can be joined to `crashes` by the common `id` column.


```r
crashes %>%
  inner_join(vehicles, by = "id") %>%
  count(vehicle) %>% 
  arrange(desc(n))
```

```
## Source: local data frame [12 x 2]
## 
##               vehicle      n
##                (fctr)  (int)
## 1                 Car 728119
## 2            Van, ute  87927
## 3  SUV or 4x4 vehicle  48269
## 4               Truck  44305
## 5          Motorcycle  17733
## 6                  NA  16996
## 7             Bicycle  15713
## 8                 Bus   8066
## 9    Taxi or taxi van   6792
## 10              Moped   3594
## 11   Other or unknown   2043
## 12         School bus    373
```

## Objects struck

There can be many objects struck in one crash, so these are recorded in a separate
`objects_struck` dataset that can be joined to `crashes` by the common `id` column.

Q: What are more fatal, trees or lamp posts?


```r
crashes %>%
  inner_join(objects_struck, by = "id") %>%
  filter(object %in% c("Trees, shrubbery of a substantial nature"
                               , "Utility pole, includes lighting columns")
  , severity != "non-injury") %>% # non-injury crashes are poorly recorded
  count(object, severity) %>% 
  group_by(object) %>%
  mutate(percent = n/sum(n)) %>%
  select(-n) %>%
  spread(severity, percent)
```

```
## Source: local data frame [2 x 4]
## 
##                                     object      fatal   serious     minor
##                                     (fctr)      (dbl)     (dbl)     (dbl)
## 1  Utility pole, includes lighting columns 0.04432701 0.2149482 0.7407248
## 2 Trees, shrubbery of a substantial nature 0.06742092 0.2459016 0.6866774
```

A: Trees (Don't worry, I know it's harder than that.)

## Causes

Causes can be joined either to the `crashes` dataset (by the common `id`
column), or to the `vehicles` dataset (by both of the commont `id` and
`vehicle_id`) columns.

The main cause groups are given in the `causes_category` column.


```r
crashes %>%
  inner_join(causes, by = "id") %>%
  group_by(cause_category, id) %>%
  tally %>%
  group_by(cause_category) %>%
  summarize(n = n()) %>%
  arrange(desc(n)) %>%
  mutate(cause_category = factor(cause_category, levels = cause_category)) %>%
  ggplot(aes(cause_category, n)) + 
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = .5))
```

![plot of chunk unnamed-chunk-9](figure/unnamed-chunk-9-1.png) 

That's odd -- where are speed, alcohol, and restraints?  They're given in `cause_subcategory`.


```r
causes %>% 
  filter(cause_subcategory == "Too fast for conditions") %>%
  count(cause) %>% 
  arrange(desc(n))
```

```
## Source: local data frame [8 x 2]
## 
##                                cause     n
##                               (fctr) (int)
## 1                          Cornering 37861
## 2                        On straight 10196
## 3                                 NA  7119
## 4        To give way at intersection  1658
## 5           At temporary speed limit  1010
## 6              At crash or emergency    55
## 7       Approaching railway crossing    44
## 8 When passing stationary school bus    37
```

There's nothing there about speed limit violations, because it's impossible to tell what speed a
vehicle was going at when it crashed.

More worryingly, how is "Alcohol test below limit" a cause for a crash?
Hopefully they filter those out when making policy decisions.


```r
levels(causes$cause) <-                # Wrap facet labels
  str_wrap(levels(causes$cause), 13)
crashes %>%
  inner_join(causes, by = "id") %>%
  filter(cause_subcategory %in% c("Alcohol or drugs")) %>%
  group_by(cause, id) %>%
  tally %>%
  group_by(cause) %>%
  summarize(n = n()) %>%               # This extra step deals with many causes per crash
  arrange(desc(n)) %>%
  mutate(cause= factor(cause, levels = cause)) %>%
  ggplot(aes(cause, n)) + 
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = .5))
```

![plot of chunk unnamed-chunk-11](figure/unnamed-chunk-11-1.png) 

```r
rm(causes)                             # Because we messed up the factor levels
```

This time, join `causes` to both `vehicles` and `crashes` to assess the
drunken cyclist menace.


```r
crashes %>%
  filter(severity == "fatal") %>%
  select(id) %>%
  inner_join(vehicles, by = "id") %>% 
  filter(vehicle == "Bicycle") %>%
  inner_join(causes, by = c("id", "vehicle_id")) %>% 
  count(cause) %>%
  arrange(desc(n))
```

```
## Source: local data frame [55 x 2]
## 
##                                                                  cause
##                                                                 (fctr)
## 1  Behind when changing lanes position or direction (includes U-turns)
## 2                                                                   NA
## 3          When required to give way to traffic from another direction
## 4                                                Wandering or wobbling
## 5                                                     At Give Way sign
## 6                           Cyclist or M/cyclist wearing dark clothing
## 7                                        Driving or riding on footpath
## 8                                             On left without due care
## 9                              When pulling out or moving to the right
## 10                                                 At steady red light
## ..                                                                 ...
## Variables not shown: n (int)
```

I think we all know what "Wandering or wobbling" means.


NEWS.md
# nzcrash 0.0.1

* The following vehicle codes are now parsed and no longer appear in `vehicles$vehicle` as `NA`:

    ```
    E = pedestrian
    K = skateboard, in-line skater etc
    Q = equestrian
    H = wheeled pedestrian (wheelchairs etc)
    O = other or unknown.
    ```

* `inst/extdata/raw2clean.R` has been fixed to work with the latest version of
  `purrr`.
* Stray pdf files have been removed from the R directory.
* Some typos have been corrected in the vignette and README.


inst/extdata/raw2clean.R
library(curl)
library(rvest)
library(dplyr)
library(tidyr)
library(purrr)
library(magrittr)
library(readr)
library(stringr)
library(lubridate)
library(sp)
library(devtools)

# Define constants ------------------------------------------------------------#


nztz <- "Pacific/Auckland"             # Time zone
nzorigin <- ymd(origin, tz = nztz)

wgsproj <- "+init=epsg:4326"           # Google maps projection
nztmproj <- "+init=epsg:2193"          # Source projection

download_dir <- tempdir()
# download_dir <- "../../../nzcrash_dev/downloads"  # Debug

home_url <- "http://www.nzta.govt.nz"  # Note, no trailing slash
source_url <- "http://www.nzta.govt.nz/resources/crash-analysis-system-data/"

data_dictionary_url <- "http://www.nzta.govt.nz/assets/resources/guide-to-coded-crash-reports/docs/guide-to-coded-crash-reports.pdf"
tcr_url <- "http://www.nzta.govt.nz/assets/resources/traffic-crash-reports/docs/traffic-crash-reports.pdf"

col_names <- c("TLA NAME",
               "CRASH ROAD",
               "CRASH DIST",
               "CRASH DIRN",
               "INTSN",
               "SIDE ROAD",
               "CRASH ID",
               "CRASH DATE",
               "CRASH DOW",
               "CRASH TIME",
               "MVMT",
               "VEHICLES",
               "CAUSES",
               "OBJECTS STRUCK",
               "ROAD CURVE",
               "ROAD WET",
               "LIGHT",
               "WTHRa",
               "JUNC TYPE",
               "TRAF CTRL",
               "ROAD MARK",
               "SPD LIM",
               "CRASH FATAL CNT",
               "CRASH SEV CNT",
               "CRASH MIN CNT",
               "PERS AGE1",
               "PERS AGE2",
               "EASTING",
               "NORTHING",
               "DUMMY")                 # Dummy column for trailing commas

col_names_nice <- c("region",
                    "road",
                    "distance_from_landmark",
                    "direction_from_landmark",
                    "landmark",
                    "side_road",
                    "id",
                    "date",
                    "weekday",
                    "time",
                    "movement",
                    "vehicles",
                    "causes",
                    "objects_struck",
                    "curvature",
                    "wetness",
                    "light",
                    "weather",
                    "junction_type",
                    "traffic_control",
                    "road_marking",
                    "speed_limit",
                    "fatalities",
                    "severe_injuries",
                    "minor_injuries",
                    "pedestrian_age",
                    "cyclist_age",
                    "easting",
                    "northing")

col_types <- "cciccciccccccccccccccciiiiiiic"

# Download metadata -----------------------------------------------------------#

curl_download(data_dictionary_url, destfile = "./data-dictionary.pdf")
curl_download(tcr_url, destfile = "./traffic-crash-report-guide.pdf")

# Obtain urls of csv files ----------------------------------------------------#

urls <- 
  html(source_url) %>% 
  html_nodes(".publication__sections--link") %>% # Get the links
  html_attrs %>%                       # Get the metadata
  zip(.simplify = TRUE) %>%                            # Convert from a list to a data.frame
  as_data_frame %>%
  select(-class) %>%                   # Drop unnecessary metadata
  rename(url = href) %>%               # Nice column names
  mutate(url = paste0(home_url, url) # Prepend parent to relative url
         , title = str_sub(title, -4)  # Only keep the year of the title
         , path = paste0(download_dir, "/", title)) # Download path

# Download csv files ----------------------------------------------------------#

urls %>%
  rowwise %>%
  do(result = curl_download(.$url, .$path))

# Read csv files --------------------------------------------------------------#

crashes <- 
  data_frame(path = paste0(download_dir, "/", list.files(download_dir))) %>%
  rowwise %>%
  do(data = read_csv(.$path
                     , skip = 1
                     , col_names = col_names
                     , col_types = col_types))

# Bind all csv datasets together
crashes %<>% .[[1]] %>% bind_rows

# Drop dummy column
crashes %<>% select(-DUMMY)

# Make nice column names -------------------------------------------------------#

crashes %<>% set_colnames(col_names_nice)

# Make nice dates --------------------------------------------------------------#

crashes %<>%
  mutate(date = dmy(date, tz = nztz)
         , time = nzorigin + hm(paste(str_sub(time, , -3L), str_sub(time, 3L)))
         , datetime = date + hours(hour(time)) + minutes(minute(time))) %>%
  select(-weekday)

# Geo-locate ------------------------------------------------------------------#

crashes_geo <- 
  crashes %>%
  filter(!is.na(easting) & 
         !is.na(northing) & 
         easting !=0 &
         northing != 0) %>%
  as.data.frame

coordinates(crashes_geo) <- ~ easting + northing
proj4string(crashes_geo) <- nztmproj

crashes_geo %<>% 
  spTransform(CRS(wgsproj)) %>%
  as.data.frame %>%
  tbl_df %>%
  select(id, easting, northing)

crashes %<>% 
  select(-easting, -northing) %>%
  left_join(crashes_geo)

rm(crashes_geo)

# Normalise -------------------------------------------------------------------#

# Movements
crashes %<>%
  separate(movement, c("movement_row", "movement_column"), 1, extra = "drop")
# street and direction of vehicle 1 are added later
# Join movements to a descriptive table
movement_codes <- 
  read_csv("./movement_codes.csv") %>%
    mutate(movement_category = factor(movement_category, levels = movement_category)
           , movement_description = factor(movement_description, levels = movement_description))
crashes %<>% 
  left_join(movement_codes, by = c("movement_row", "movement_column")) %>%
  select(-movement_row, -movement_column)
# Reorder factors by count
m1 <- 
  crashes %>%
  count(movement_category, sort = TRUE) %>%
  .$movement_category
crashes %<>%
  mutate(movement_category = factor(movement_category, levels = m1))
m2 <- 
  crashes %>%
  group_by(movement_category, movement_description) %>%
  tally %>%
  arrange(movement_category, desc(n)) %>%
  mutate(combination = paste0(movement_category, movement_description))
crashes %<>%
  mutate(movement_description = factor(paste0(movement_category, movement_description)
                                , levels = m2$combination
                                , labels = m2$movement_description))
rm(m1, m2, movement_codes)

# Vehicles 
vehicles <- 
  crashes %>%
  select(id, vehicles) %>%
  separate(vehicles, c("a", "dummy1", "dummy2", "b", "c", "d"), 1:5
           , extra = "drop", remove = FALSE) %>% # Can't remove 'vehicles' until streetdirection has been obtained
  gather(key = "vehicle", value = "type", a, b, c, d) %>%
  select(id, vehicle, type) %>%
  filter(type != "") %>%
  arrange(id, vehicle) 
# Can't remove 'vehicles' column of 'crashes' until streetdirection has been obtained
vehicles %<>% 
  mutate(vehicle_id = as.character(vehicle)
         , vehicle = factor(type
                       , levels = c("C", "V", "X", "B", "L", "4", "T", "M", "P",
                                    "S", "E", "K", "Q", "H", "O") # U is not defined in the data dictionary
                       , labels = c("Car", "Van, ute"
                                    , "Taxi or taxi van", "Bus"
                                    , "School bus"
                                    , "SUV or 4x4 vehicle", "Truck"
                                    , "Motorcycle", "Moped", "Bicycle"
                                    , "Pedestrian", "Skateboard, in-line skater, etc."
                                    , "Equestrian", "Wheeled pedestrian (wheelchairs, etc.)"
                                    , "Other or unknown"))) %>%
select(id, vehicle_id, vehicle)

# Street and direction of vehicle 1 in movement 
crashes %<>%
  separate(vehicles, c("dummy1", "direction", "street", "dummy2"), 1:3, extra = "drop") %>% 
  select(-dummy1, -dummy2)

# Causes
maxcauses <- max(str_count(crashes$causes, " ")) # space-delimited string
causes <- 
  crashes %>%
  select(id, causes) %>%
  separate(causes, 1:maxcauses, extra = "drop")
causes[causes == ""] <- NA             # trailing spaces create empty strings
causes %<>%
  gather_("index", "value", as.character(1:maxcauses)) %>%
  filter(!is.na(value)) %>%
  separate(value, c("code", "vehicle_id"), 3, extra = "drop") %>%
  mutate(index = as.integer(index)
         , code = as.integer(code)
         , vehicle_id = str_to_lower(vehicle_id))
causes[causes == ""] <- NA             # 800-900-series causes are environmental (not associated with vehicles)
crashes %<>% select(-causes)
# Join to code descriptions
causes_table <- read.csv("./causes.csv", sep = "|")
causes %<>% 
  left_join(causes_table)
rm(causes_table, maxcauses)

# Objects struck
objects_struck <- 
  crashes %>%
  select(id, objects_struck) %>%
  separate(objects_struck, 1:3, 1:2, extra = "drop") %>%
  gather_("object_order", "object", as.character(1:3))
objects_struck[objects_struck == ""] <- NA # separate fills with blanks
objects_struck %<>%
  filter(!is.na(object)) %>%
  arrange(id, object_order)
crashes %<>% select(-objects_struck)
objects_struck %<>%
  mutate(object_order = as.integer(object_order)
         , object = factor(object
                           , levels = LETTERS[c(1:14, 16:20, 22:26)]
                           , labels = c("Driven or accompanied animals, i.e. under control"
                                        , "Bridge abutment, handrail or approach, includes tunnels"
                                        , "Upright cliff or bank, retaining walls"
                                        , "Debris, boulder or object dropped from vehicle"
                                        , "Over edge of bank"
                                        , "Fence, letterbox, hoarding etc."
                                        , "Guard or guide rail (including median barriers)"
                                        , "House or building"
                                        , "Traffic island or median strip"
                                        , "Public furniture, e.g. phone boxes, bus shelters, signal controllers, etc."
                                        , "Kerb, when directly contributing to incident"
                                        , "Landslide, washout or floodwater"
                                        , "Parked motor vehicle"
                                        , "Train"
                                        , "Utility pole, includes lighting columns"
                                        , "Broken down vehicle, workmen's vehicle, taxis picking up, etc."
                                        , "Roadwork signs or drums, holes and excavations, etc."
                                        , "Traffic signs or signal bollards"
                                        , "Trees, shrubbery of a substantial nature"
                                        , "Ditch"
                                        , "Wild animal, strays, or out of control animals"
                                        , "Other"
                                        , "Objects thrown at or dropped onto vehicles"
                                        , "Into water, river or sea")))

# Light (needn't be separate from crashes)
crashes %<>%
  separate(light, c("light_ambient", "light_artificial"), 1, extra = "drop") %>%
  mutate(light_ambient = ifelse(light_ambient == " ", NA, light_ambient) # separate fills with blanks
         , light_artificial = ifelse(light_artificial == " ", NA, light_artificial))
crashes %>%
  select(light_ambient, light_artificial)

# Weather (needn't be separate from crashes)
crashes %<>%
  separate(weather, paste0("weather_", 1:2), 1, extra = "drop") %>%
  mutate(weather_1 = ifelse(weather_1 == " ", NA, weather_1) # separate fills with blanks
         , weather_2 = ifelse(weather_2 == " ", NA, weather_2))

# Severity (worst injury) of crash (needn't be separate from crashes)
crashes %<>%
  mutate(severity  = ifelse(fatalities >= 1, "fatal"
                            , ifelse(severe_injuries >= 1, "serious"
                                     , ifelse(minor_injuries >= 1, "minor", "non-injury")))
         , severity = factor(severity, levels = c("fatal"
                                                  , "serious"
                                                  , "minor"
                                                  , 'non-injury')))

# Make certain columns factors for empty levels
crashes %<>%
  mutate(direction_from_landmark = ifelse(direction_from_landmark == " "
                                          , NA
                                          , direction_from_landmark)
         , direction_from_landmark = factor(direction_from_landmark
                                            , levels = c("N", "E", "S", "W")
                                            , labels = c("North", "East"
                                                         , "South", "West"))
         , landmark = ifelse(landmark == " " , NA , landmark)
         , landmark = factor(landmark
                             , levels = c("I", "A")
                             , labels = c("Intersection", "Landmark"))
         , direction = factor(direction
                              , levels = c("N", "E", "S", "W", "U")
                              , labels = c("North", "East", "South", "West"
                                           , "Unknown"))
         , street = factor(street
                           , levels = c("1", "2", "C")
                           , labels = c("First", "Second", "C"))
         , curvature = factor(curvature
                              , levels = c("R", "E", "M", "S")
                              , labels = c("Straight", "Easy", "Medium"
                                           , "Severe"))
         , wetness = factor(wetness
                            , levels = c("D", "W", "I")
                            , labels = c("Dry", "Wet", "Ice or snow"))
         , light_ambient = factor(light_ambient
                                  , levels = c("B", "O", "T", "D")
                                  , labels = c("Bright sun", "Overcast"
                                               , "Twilight", "Dark"))
         , light_artificial = factor(light_artificial
                                     , levels = c("O", "F", "N")
                                     , labels = c("Street lights on"
                                                  , "Street lights off"
                                                  , "No street lights present"))
         , weather_1 = factor(weather_1
                              , levels = c("F", "M", "L", "H", "S")
                              , labels = c("Fine", "Mist/fog", "Light rain"
                                           , "Heavy rain", "Snow"))
         , weather_2 = factor(weather_2
                              , levels = c("F", "S")
                              , labels = c("Frost", "Strong wind"))
         , junction_type = factor(junction_type
                                  , levels = c("D", "R", "X", "T", "Y", "M")
                                  , labels = c("Driveway"
                                               , "Roundabout"
                                               , "Crossroads"
                                               , "T junction"
                                               , "Y junction"
                                               , "Multi-leg (>4)"))
         , traffic_control = factor(traffic_control
                                    , levels = c("T", "S", "G", "P", "N")
                                    , labels = c("Traffic signals"
                                                 , "Stop sign"
                                                 , "Give way sign"
                                                 , "School patrol or warden"
                                                 , "Nil"))
         , road_marking = factor(road_marking
                                 , levels = c("X", "R", "P", "L", "C", "N")
                                 , labels = c("Pedestrian crossing"
                                              , "Raised island"
                                              , "painted island"
                                              , "no passing line"
                                              , "centre line"
                                              , "Nil"))
         , speed_limit = factor(speed_limit
                                , levels = c("LSZ", "005", "010", "015", "020"
                                             , "025", "030", "040", "050"
                                             , "060", "070", "080", "090"
                                             , "100")
                                , labels = c("LSZ", "5", "10", "15", "20", "25"
                                             , "30", "40", "50", "60", "70"
                                             , "80", "90", "100")))

# Convert everything back to data.frame
crashes %<>% as.data.frame
causes %<>% as.data.frame
objects_struck %<>% as.data.frame
vehicles %<>% as.data.frame

# Save data in the package
devtools::use_data(crashes, causes, objects_struck, vehicles, overwrite = TRUE, compress = "xz")
